src/geometry/GeometryEvaluator.cc

```diff
@@ -1250,67 +1249,72 @@ static void fill_ring(std::vector<Vector3d>& ring, const Outline2d& o, double a,
    Currently, we generate a lot of zero-area triangles
 
  */
-static std::unique_ptr<Geometry> rotatePolygon(const RotateExtrudeNode& node, const Polygon2d& poly)
+static std::unique_ptr<Geometry> rotatePolygonSub(const RotateExtrudeNode& node, const Polygon2d& poly, int fragments, int fragstart, int fragend, bool flip_faces)
 {
-  if (node.angle == 0) return nullptr;
 
   PolySetBuilder builder;
   builder.setConvexity(node.convexity);
 
-  double min_x = 0;
-  double max_x = 0;
-  unsigned int fragments = 0;
-  for (const auto& o : poly.outlines()) {
-    for (const auto& v : o.vertices) {
-      min_x = fmin(min_x, v[0]);
-      max_x = fmax(max_x, v[0]);
-    }
-  }
-
-  if ((max_x - min_x) > max_x && (max_x - min_x) > fabs(min_x)) {
-    LOG(message_group::Error, "all points for rotate_extrude() must have the same X coordinate sign (range is %1$.2f -> %2$.2f)", min_x, max_x);
-    return nullptr;
-  }
 
-  fragments = (unsigned int)std::ceil(fmax(Calc::get_fragments_from_r(max_x - min_x, node.fn, node.fs, node.fa) * std::abs(node.angle) / 360, 1));
 
-  bool flip_faces = (min_x >= 0 && node.angle > 0 && node.angle != 360) || (min_x < 0 && (node.angle < 0 || node.angle == 360));
+  double fact=(node.v[2]/node.angle)*(180.0/G_PI);
```

Comment:
G_PI: That's the first time I've seen that. No idea where that's coming from, but it feels wrong.

I think the cleanest way to achieve this prior to C++20 is:
```
#define _USE_MATH_DEFINES
#include <cmath>
```